### DISCUSSION TITLE
Popup Save before exit.

### DISCUSSION THREAD

**Discussion by MikolajGodzicki** [AvaloniaUI/Avalonia#7040] [Q&A]
Hello
I'm making simple notepad for school and I can't find how to do popup window with text: "Do you want save before exit?" and two buttons: yes and no. Is there any template or tutorial how to make it?

**Comment by vadd98** [comment]
You could give a look at the ContentDialog available here https://github.com/PieroCastillo/Aura.UI or at the MessageBox here https://github.com/AvaloniaCommunity/MessageBox.Avalonia.
Otherwise, you can implement a popup from scratch following this guide https://docs.avaloniaui.net/tutorials/music-store-app/opening-a-dialog

**Comment by maxkatz6** [comment]
Window.Closing event
And ((IClassicDesktopStyleApplicationLifetime)Application.Current.ApplicationLifetime).ShutdownRequested events
